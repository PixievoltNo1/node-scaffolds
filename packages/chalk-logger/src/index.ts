import type { Mutable } from '@guanghechen/utility-types'
import { partOptionsFromArgs } from './commander'
import type { Level } from './level'
import type { ILoggerOptions } from './logger'
import { Logger } from './logger'

export * from './color'
export * from './commander'
export * from './level'
export * from './logger'

export class ChalkLogger extends Logger {
  /**
   * prefix of logger.name
   */
  protected basename: string | null = null
  /**
   * name passed into .setName()
   */
  protected divisionName: string | null = null

  constructor(options?: ILoggerOptions, args?: string[]) {
    const optionsFromArgs = partOptionsFromArgs(args || [])
    const resolvedOptions = {
      ...options,
      ...optionsFromArgs,
      flags: { ...options?.flags, ...optionsFromArgs.flags },
    }

    super(resolvedOptions)
    const basename = resolvedOptions?.name ?? ''
    this.setBaseName(basename)
  }

  /**
   * update logger's level
   * @param level
   */
  public setLevel(level: Level | null | undefined): void {
    if (level == null) return
    const self = this as Mutable<this>
    self.level = level
  }

  /**
   * update logger's name
   * @param name
   */
  public setName(name: string | null): void {
    const resolvedName: string = [this.basename, name]
      .filter((x): x is string => x != null && x.length > 0)
      .join(' ')
    const self = this as Mutable<this>
    self.name = resolvedName
    this.divisionName = name
  }

  /**
   * update basename of logger
   * @param basename
   */
  public setBaseName(basename: string | null): void {
    this.basename = basename
    this.setName(this.divisionName)
  }

  /**
   * update logger's mode
   * @param mode
   */
  public setMode(mode: 'normal' | 'loose'): void {
    const self = this as Mutable<this>
    self.mode = mode
  }
}

export default ChalkLogger
