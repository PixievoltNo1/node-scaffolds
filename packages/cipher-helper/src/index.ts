export * from './_base'
export * from './aes'
export * from './catalog'
export * from './types/catalog'
export * from './types/cipher-helper'
export * from './util/buffer'
export * from './util/key'
