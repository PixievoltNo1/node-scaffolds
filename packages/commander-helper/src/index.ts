export * from './command'
export * from './fs'
export * from './git'
export * from './merge'
export * from './option'
export * from './yarn'
