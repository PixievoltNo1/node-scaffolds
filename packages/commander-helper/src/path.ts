import path from 'path'

/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetPath
 */
export function absoluteOfWorkspace(
  workspace: string,
  targetPath?: string | null,
): string {
  if (targetPath == null) return workspace
  const absoluteDir: string = path.resolve(workspace, targetPath)
  return absoluteDir
}

/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetDir
 */
export function relativeOfWorkspace(
  workspace: string,
  targetDir: string,
): string {
  const absoluteDir = absoluteOfWorkspace(workspace, targetDir)
  const relativeDir: string = path.relative(workspace, absoluteDir)
  return path.normalize(relativeDir)
}
