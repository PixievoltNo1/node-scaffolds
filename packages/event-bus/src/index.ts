export * from './SimpleEventBus'
export * from './types'
