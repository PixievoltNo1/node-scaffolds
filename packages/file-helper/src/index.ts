export * from './big-file-helper'
export * from './types'
export * from './util/fs'
export * from './util/split-file'
export * from './util/stream'
