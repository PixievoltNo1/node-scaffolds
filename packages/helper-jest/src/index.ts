export * from './desensitizer/json'
export * from './desensitizer/types'
export * from './desensitizer/string'
export * from './mock/console'
export * from './mock/logger'
export * from './snapshot/file'
