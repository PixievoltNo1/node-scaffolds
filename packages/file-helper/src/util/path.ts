import path from 'path'

/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetPath
 */
export function absoluteOfWorkspace(workspace: string, targetPath?: string | null): string {
  if (targetPath == null) return workspace
  const filepath: string = path.resolve(workspace, targetPath)
  return path.normalize(filepath)
}

/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetPath
 */
export function relativeOfWorkspace(workspace: string, targetPath: string): string {
  const absoluteDir = absoluteOfWorkspace(workspace, targetPath)
  const filepath: string = path.relative(workspace, absoluteDir)
  return path.normalize(filepath)
}
