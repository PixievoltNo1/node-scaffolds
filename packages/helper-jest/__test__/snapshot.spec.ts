import path from 'node:path'
import url from 'node:url'
import { fileSnapshot } from '../src'

const __dirname = path.dirname(url.fileURLToPath(import.meta.url))

describe('fileSnapshot', function () {
  test('basic', function () {
    fileSnapshot(path.join(__dirname, 'fixtures'), ['data1.json'])
  })

  test('with desensitize', function () {
    fileSnapshot(path.join(__dirname, 'fixtures'), ['data1.json'], text =>
      text
        .replace(/1/g, '-one')
        .replace(/2/g, '-two')
        .replace(/3/g, '-three')
        .replace(/4/g, '-four'),
    )
  })
})
