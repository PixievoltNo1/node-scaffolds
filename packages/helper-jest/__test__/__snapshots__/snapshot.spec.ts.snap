// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fileSnapshot basic: data1.json 1`] = `
"{
  \\"name\\": \\"test data\\",
  \\"value\\": \\"somethings\\",
  \\"array\\": [
    \\"this is an array\\",
    \\"this is an array\\",
    \\"this is an array\\",
    \\"this is an array\\"
  ],
  \\"object\\": {
    \\"key\\": \\"value\\",
    \\"key2\\": \\"value2\\",
    \\"key3\\": \\"value3\\",
    \\"key4\\": \\"value4\\"
  }
}
"
`;

exports[`fileSnapshot with desensitize: data1.json 1`] = `
"{
  \\"name\\": \\"test data\\",
  \\"value\\": \\"somethings\\",
  \\"array\\": [
    \\"this is an array\\",
    \\"this is an array\\",
    \\"this is an array\\",
    \\"this is an array\\"
  ],
  \\"object\\": {
    \\"key\\": \\"value\\",
    \\"key-two\\": \\"value-two\\",
    \\"key-three\\": \\"value-three\\",
    \\"key-four\\": \\"value-four\\"
  }
}
"
`;
